File: resize-logo.js
const fs = require('fs');
const path = require('path');

// Check if sharp is installed, if not, provide instructions
try {
  const sharp = require('sharp');
  
  const publicDir = path.join(__dirname, 'public');
  const logoPath = path.join(publicDir, 'logo_pwa.png');
  
  if (!fs.existsSync(logoPath)) {
    console.error('Error: logo_pwa.png not found in public folder');
    process.exit(1);
  }
  
  console.log('Resizing logo_pwa.png...');
  
  // Create 192x192 version
  sharp(logoPath)
    .resize(192, 192, { fit: 'cover' })
    .png()
    .toFile(path.join(publicDir, 'logo_pwa_192.png'))
    .then(() => console.log('✓ Created logo_pwa_192.png'))
    .catch(err => console.error('Error creating 192px logo:', err));
  
  // Create 512x512 version
  sharp(logoPath)
    .resize(512, 512, { fit: 'cover' })
    .png()
    .toFile(path.join(publicDir, 'logo_pwa_512.png'))
    .then(() => console.log('✓ Created logo_pwa_512.png'))
    .catch(err => console.error('Error creating 512px logo:', err));
    
} catch (error) {
  console.error('Sharp not installed. Installing it now...');
  require('child_process').execSync('npm install sharp', { stdio: 'inherit' });
}

File: resize_logo.py
from PIL import Image
import os

public_dir = 'public'
logo_path = os.path.join(public_dir, 'logo_pwa.png')

if not os.path.exists(logo_path):
    print(f"Error: {logo_path} not found")
    exit(1)

try:
    # Open the original logo
    img = Image.open(logo_path)
    
    # Create 192x192 version
    img_192 = img.resize((192, 192), Image.Resampling.LANCZOS)
    img_192.save(os.path.join(public_dir, 'logo_pwa_192.png'))
    print("✓ Created logo_pwa_192.png")
    
    # Create 512x512 version
    img_512 = img.resize((512, 512), Image.Resampling.LANCZOS)
    img_512.save(os.path.join(public_dir, 'logo_pwa_512.png'))
    print("✓ Created logo_pwa_512.png")
    
    print("Logo resizing complete!")
    
except Exception as e:
    print(f"Error: {e}")
    exit(1)
